### DISCUSSION TITLE
Git Environments - Ability to Save As and create a Duplicate Environment

### DISCUSSION THREAD

**Discussion by ppmgitrepo** [community/community#120102] [Enterprise]
**The problem**
When we define an Environment in Git it can have 30+ secrets and another 20+ Environment variables defined.
Identical environments need to be created for SIT, UAT, Prod, Dev , Conversions etc..

Without the ability to do a Save As and create a copy of the Environment we need to type in everything from scratch which is a tedious process.

**The solution**
Please give the ability to select an Environment and do a Save as > Another Environment.
This way all the Secret variables and Environment variables get created, it is just a matter of changing their values.

**Comment by leon** [comment]
Another solution would be to allow creating secrets / environment variables from .env files
that way you can simply upload a full file with all secrets at the same time.

**Reply by 9Rune5** [reply]
In that case you can simply use the gh command line utility?

**Reply by leon** [reply]
Probably, but it would be a lot easier being able to do so via the UI.

**Reply by ppmgitrepo** [reply]
Yes, this too would work. But a better option would be is to allow a direct save as at the Environment Level.

**Comment by ghost** [comment]
This needs some solution. When we are testing multiple environments and have a new branch and each branch may have some updates to configuration/env variables, then we should be able to make a branch based deployment for our sandbox environments and just copy a previous environment as a starting point.

Come on, Github. This can't be that hard of a change.

**Comment by gustavovalverde** [comment]
If it helps, I created a bash script for myself to accomplish this using `gh` CLI, it might be useful for others. https://gist.github.com/gustavovalverde/b0e566c2da331ae51116db9f74d30583

It allows creating environments with their variables using an `.env` file, and copying variables from one environment to another.
